core/src/main/java/edu/wpi/grip/core/http/GripServer.java

```diff
@@ -0,0 +1,404 @@
+        }
+        contextPaths.add(path);
+        server.createContext(path, makeContext(path));
+    }
+
+    private HttpHandler makeContext(final String path) {
+        return he -> {
+            final Headers headers = he.getResponseHeaders();
+            final String requestMethod = he.getRequestMethod().toUpperCase();
+            final Map<String, List<String>> requestParameters = getRequestParameters(he.getRequestURI());
+            switch (requestMethod) {
+                case METHOD_GET:
+                    headers.set(HEADER_CONTENT_TYPE, String.format("application/json; charset=%s", CHARSET));
+                    if (getHandlers.containsKey(path)) {
+                        final String responseBody = getHandlers.get(path).createResponse(requestParameters);
+                        final byte[] rawResponseBody = responseBody.getBytes(CHARSET);
+                        he.sendResponseHeaders(STATUS_OK, rawResponseBody.length);
+                        he.getResponseBody().write(rawResponseBody);
+                        he.getResponseBody().close();
+                    } else {
+                        he.sendResponseHeaders(STATUS_OK, NO_RESPONSE_LENGTH);
+                    }
+                    break;
+                case METHOD_POST:
+                    final byte[] bytes = readBytes(he);
+                    boolean success = true;
+                    for (PostHandler handler : postHandlers.get(path)) {
+                        success &= handler.convert(bytes);
+                    }
+                    if (success) {
+                        he.sendResponseHeaders(STATUS_OK, NO_RESPONSE_LENGTH);
+                    } else {
+                        he.sendResponseHeaders(STATUS_INTERNAL_ERROR, NO_RESPONSE_LENGTH);
+                    }
+                    break;
+                case METHOD_OPTIONS:
+                    headers.set(HEADER_ALLOW, ALLOWED_METHODS);
+                    he.sendResponseHeaders(STATUS_OK, NO_RESPONSE_LENGTH);
+                    break;
+                default:
+                    headers.set(HEADER_ALLOW, ALLOWED_METHODS);
+                    he.sendResponseHeaders(STATUS_METHOD_NOT_ALLOWED, NO_RESPONSE_LENGTH);
+                    break;
+            }
+        };
+    }
+
+    /**
+     * Adds a data supplier for data with the given name. Each name can only be
+     * associated with a single supplier at a time, so this will remove the previous
+     * supplier (if one exists).
+     *
+     * @param name     the name of the data
+     * @param supplier the supplier of the data
+     */
+    public void addDataSupplier(String name, Supplier<?> supplier) {
+        dataSuppliers.put(name, supplier);
+    }
+
+    /**
+     * Checks if the given name is associated with a supplier.
+     *
+     * @param name the name to check
+     * @return true if the name is associated with a supplier, false if not
+     */
+    public boolean hasDataSupplier(String name) {
```

Comment:
If this is only used in tests then make it package private and mark it `@VisibleForTesting`
but ideally you should only really test things you need to exist as a public method.